### PR TITLE
FIX : use random state in infomax and make it default to None

### PR DESCRIPTION
**Pull request by agramfort**

### CODE REVIEW

**Comment by agramfort** [comment]
closes #2084

ping @dengemann

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def __init__(self, n_components=None, max_pca_components=None,
         self.max_pca_components = max_pca_components
         self.n_pca_components = n_pca_components
         self.ch_names = None
-        self.random_state = random_state if random_state is not None else 42
+        self.random_state = random_state
```
Comment (Eric89GXL) [inline mne/preprocessing/ica.py]: unfortunately this one is not backward compatible :( Unless this is considered a bug...

Comment (agramfort) [inline mne/preprocessing/ica.py]: Either we fixed the seed or we don't. Currently seed was used in PCA but not in infomax

Comment (Eric89GXL) [inline mne/preprocessing/ica.py]: I agree about seeding, but the default used to be fixed to `42` (in some places) and now it defaults to `None`

Comment (agramfort) [inline mne/preprocessing/ica.py]: I am fine with setting the seed @dengemann is it the behavior you would expect?

Comment (dengemann) [inline mne/preprocessing/ica.py]: Yes, absolutely, this is a bug.

**Comment by dengemann** [comment]
Thanks @agramfort !

**Comment by teonlamont** [comment]
thanks @agramfort!

**Pull request merged**